### DISCUSSION TITLE
Dvorak keyboard layout

### DISCUSSION THREAD

**Discussion by JoeStrout** [rsta2/circle#433] [Show and tell]
Day 2 of using Circle, and I'm loving it.  But I type in Dvorak; typing in QWERTY is slow and painful for me.  So I made a **Dvorak layout**, and thought I'd share.

[keymap_dv.h.zip](https://github.com/rsta2/circle/files/14393143/keymap_dv.h.zip)

To use:
1. Unzip and place the above keymap_dv.h file into lib/input.
2. Edit keymap.cpp, lines 101-130, to include the new map, like so:

```cpp
const u16 CKeyMap::s_DefaultMap[][PHY_MAX_CODE+1][K_CTRLTAB+1] =
{
	{
		#include "keymap_de.h"
	}, {
		#include "keymap_dv.h"
	}, {
		#include "keymap_es.h"
	}, {
		#include "keymap_fr.h"
	}, {
		#include "keymap_it.h"
	}, {
		#include "keymap_uk.h"
	}, {
		#include "keymap_us.h"
	}
};

const char *CKeyMap::s_MapDirectory[] =		// same (alphabetical) order as in s_DefaultMap[]
{
	"DE",
	"DV",
	"ES",
	"FR",
	"IT",
	"UK",
	"US",
	0
};
```

3. Rebuild (a `./makeall` ought to do it).
4. Add `keymap=DV` to your _cmdline.txt_ file.

Enjoy!

**Comment by rsta2** [comment]
Thanks for contributing this keyboard layout! I added it on the *develop* branch. It will be part of the upcoming release 46.
